clib and python: Support logging

cathay4t opened the pull request:
Enable logging in C and python binding for all `nmstate` and `nispor`
logs. The logs are stored in json format as `Vec<LogEntry>`.

The C binding is storing logs in memory and drain it after every public
function call.

Python binding just read logs from C binding and send to python
`logging` module.

To test it, you may run this python script:

```py
import sys
import logging
import libnmstate
import yaml

logging.basicConfig(stream=sys.stderr, level=logging.INFO)
libnmstate.show()
```

cathay4t commented:
Checked locally, the pytest now collect rust logs when failure happens.

qinqon commented:
@cathay4t Is this going to work with calls being done in parallel ? The memory is going to be share between them so we will show logs from the other calls and also drain them too.

I se that for integration test is not an issue but for the golang binding the locking, drain and lack of isolation it may be a problem, can you include a mechanism to disable it ?

cathay4t commented:
~~I will introduce a transaction ID for each function. Thanks!~~

I will try to find a fix for this.

cathay4t commented:
New patch uploaded. To handle multi-threading:
 * Each C binding call will store a `SystemTime::now()` at beginning of function and register as consumer to logger.
 * Then `MemoryLogger.drain()` only clear logs when calling from last consumer.
 * The log is filtered by the time function stored.
 
I don't like have locking at `nmstate` level, user should do it at their code, so mixing logs between functions is expected.

cathay4t commented:
~~Haven't test the code yet.~~

cathay4t committed:
clib and python: Support logging.

Enable logging in C and python binding for all `nmstate` and `nispor`
logs. The logs are stored in json format as `Vec<LogEntry>`.

The C binding is storing logs in memory and drain it after every public
function call.

Python binding just read logs from C binding and send to python
`logging` module.

To test it, you may run this python script:

```py
import sys
import logging
import libnmstate
import yaml

logging.basicConfig(stream=sys.stderr, level=logging.INFO)
libnmstate.show()
```

When running in multithread, logs from other process will also included.
Library consumer should ensure only one thread is doing the apply
action.

Signed-off-by: Gris Ge <fge@redhat.com>
